merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/AllDefinedMethodMaker.java

```diff
@@ -61,7 +61,9 @@ public MethodSpec makeInstantiateMethod() {
 
     methodBuilder = makeArgumentAssignments(methodBuilder, (builder, parameter) -> builder
         .addStatement(
-            "template.$L = this.mapper_$L.deserializeValue($L.getValue())$W.getSuccessOrThrow(failure -> new $T(\"$L\", failure))",
+            "$L = $T.ofNullable(template.$L = this.mapper_$L.deserializeValue($L.getValue())$W.getSuccessOrThrow(failure -> new $T(\"$L\", failure)))",
```

Comment:
🔵 As you're well aware - this assumption of `null` === "missing parameter" doesn't work if parameters are primitive types. We should make a ticket to handle those correctly (or, alternatively, Deprecate "Classic" style in favor of record style activities)

Reply:
Deprecation has my vote 👀 If we had records when Aerie was started, I doubt we ever would have supported anything else.

Reply:
The one thing you can't do with records is do subclassing, because the Record superclass already fills the one inheritance slot Java gives us. I don't think this is something people should be doing with activities, but it would be worth scanning for such uses anyway.

Reply:
Just created this ticket to look into deprecating classic style activities and/or warning the mission modeler when defining a `@Parameter` of a primitive type with no explicit assignment statement: https://jira.jpl.nasa.gov/browse/AERIE-2122.